include/Graph/Graph.h

```diff
@@ -55,8 +55,13 @@ void Graph<Node>::insert(Node *Src, Node *Dest, int Left, int Right) {
     }
   } else if (Left == 2 && Right == 1) {
     for (auto Src : this->getPointee(Src)) {
-      for (auto Dest : this->getPointee(Dest)) {
-        this->insert(Src, Dest);
+      if(Dest->isNull()){
```

Comment:
Points-to graph need not know anything about nullptr. We should treat nullptr as a location in the heap and should keep redirection level 0 for it.